crates/pallets/starknet/src/lib.rs

```diff
@@ -295,17 +372,97 @@ pub mod pallet {
-        /// * Check if the contract class hash is known.
         fn _associate_contract_class(
             contract_address: ContractAddress,
             contract_class_hash: ContractClassHash,
         ) -> Result<(), DispatchError> {
+            // Check if the contract address is already associated with a contract class hash.
+            ensure!(
+                !ContractClassHashes::<T>::contains_key(contract_address),
+                Error::<T>::ContractClassHashAlreadyAssociated
+            );
+
+            // Check if the contract class hash is known.
+            ensure!(ContractClassHashes::<T>::contains_key(contract_class_hash), Error::<T>::ContractClassHashUnknown);
+
             ContractClassHashes::<T>::insert(contract_address, contract_class_hash);
+
             Ok(())
         }
+
+        /// Create a state reader.
+        ///
+        /// # Returns
+        ///
+        /// The state reader.
+        fn create_state_reader() -> CachedState<DictStateReader> {
+            let address_to_class_hash: HashMap<StarknetContractAddress, ClassHash> = ContractClassHashes::<T>::iter()
+                .map(|(key, value)| {
+                    (
+                        StarknetContractAddress::try_from(StarknetStarkFelt::new(key).unwrap()).unwrap(),
+                        ClassHash(StarknetStarkFelt::new(value).unwrap()),
+                    )
+                })
+                .collect();
+
+            let address_to_nonce: HashMap<StarknetContractAddress, StarknetNonce> = Nonces::<T>::iter()
+                .map(|(key, value)| {
+                    (
+                        StarknetContractAddress::try_from(StarknetStarkFelt::new(key).unwrap()).unwrap(),
+                        StarknetNonce(StarknetStarkFelt::new(value.into()).unwrap()),
+                    )
+                })
+                .collect();
+
+            let storage_view: HashMap<StarknetContractStorageKey, StarknetStarkFelt> = StorageView::<T>::iter()
+                .map(|(key, value)| {
+                    (
+                        (
+                            StarknetContractAddress::try_from(StarknetStarkFelt::new(key.0).unwrap()).unwrap(),
+                            StorageKey::try_from(StarknetStarkFelt::new(key.1.into()).unwrap()).unwrap(),
+                        ),
+                        StarknetStarkFelt::new(value.into()).unwrap(),
+                    )
+                })
+                .collect();
+
+            // let class_hash_to_class: ContractClassMapping = ContractClasses::<T>::iter().map(|(key, value)| {
+            // 	let class_hash = ClassHash(StarknetStarkFelt::new(key).unwrap());
+            // 	let contract_class = ContractClass::try_from(value).unwrap();
+            // 	(class_hash, contract_class)
+            // }).collect();
+            // TODO: remove this when declare is implemented
+            let class_hash_to_class: ContractClassMapping = HashMap::from([
+                (
```

Comment:
we need to think how to iterate with some mocked/partial features going forward, as well as proper testing

Reply:
This is just temporary, as once we have declare transactions we can remove it.
Proper mocking/testing is done through the genesis config initialisation in `mock.rs` and `test.rs` like it's done for contract addresses and class hashes.